samples/user-interface/windowmanager/src/main/java/com/example/platform/ui/windowmanager/MediaPlayerActivity.kt

```diff
@@ -0,0 +1,162 @@
+ * limitations under the License.
+ */
+
+package com.example.platform.ui.windowmanager
+
+import android.os.Build
+import androidx.appcompat.app.AppCompatActivity
+import android.os.Bundle
+import android.view.View
+import android.view.WindowInsets
+import android.view.WindowInsetsController
+import android.view.WindowManager
+import androidx.constraintlayout.widget.ConstraintLayout
+import androidx.lifecycle.Lifecycle
+import androidx.lifecycle.lifecycleScope
+import androidx.lifecycle.repeatOnLifecycle
+import androidx.media3.common.MediaItem
+import androidx.media3.common.util.UnstableApi
+import androidx.media3.exoplayer.ExoPlayer
+import androidx.media3.exoplayer.trackselection.DefaultTrackSelector
+import androidx.window.layout.FoldingFeature
+import androidx.window.layout.WindowInfoTracker
+import androidx.window.layout.WindowLayoutInfo
+import com.example.platform.ui.windowmanager.databinding.ActivityMediaPlayerBinding
+import com.example.platform.ui.windowmanager.util.foldPosition
+import com.example.platform.ui.windowmanager.util.isTableTopMode
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlinx.coroutines.launch
+
+@UnstableApi class MediaPlayerActivity : AppCompatActivity() {
+    private var player: ExoPlayer? = null
+    private var playWhenReady = true
+    private var currentMediaIndex = 0
+    private var playbackPosition: Long = 0
+
+    private lateinit var binding: ActivityMediaPlayerBinding
+
+    @ExperimentalCoroutinesApi
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+
+        binding = ActivityMediaPlayerBinding.inflate(layoutInflater)
+        setContentView(binding.root)
+
+        lifecycleScope.launch(Dispatchers.Main) {
+            lifecycle.repeatOnLifecycle(Lifecycle.State.STARTED) {
+                WindowInfoTracker.getOrCreate(this@MediaPlayerActivity)
+                    .windowLayoutInfo(this@MediaPlayerActivity)
+                    .collect { newLayoutInfo ->
+                        onLayoutInfoChanged(newLayoutInfo)
+                    }
+            }
+        }
+    }
+
+    override fun onStart() {
+        super.onStart()
+        initializePlayer()
+    }
+
+    override fun onResume() {
+        super.onResume()
+        hideSystemUI()
+    }
+
+    override fun onStop() {
+        super.onStop()
+        releasePlayer()
+    }
+
+    private fun initializePlayer() {
+        if (player == null) {
+            val trackSelector = DefaultTrackSelector(this)
+            trackSelector.setParameters(
+                trackSelector.buildUponParameters().setMaxVideoSizeSd()
+            )
+            player = ExoPlayer.Builder(this)
+                .setTrackSelector(trackSelector)
+                .build()
+        }
+        binding.playerView.player = player
+        val mediaItem = MediaItem.fromUri(getString(R.string.media_url_mp4))
+        player?.setMediaItem(mediaItem)
```

Comment:
```suggestion
        player?.let { ->
```

should we make the null check only once?

Reply:
Yes. Done